subsys/net/ip/net_context.c

```diff
@@ -617,7 +641,7 @@ int net_context_listen(struct net_context *context, int backlog)
 #endif /* CONFIG_NET_OFFLOAD */
 
 	if (net_tcp_listen(context) >= 0) {
-		return 0;
+		return net_tls_connect(context, true);
```

Comment:
Why do we have a function named `net_tls_connect`, which doesn't necessarily perform "TLS connect", in which regard it differs from othe `net_tls_*` functions. For example, `net_tls_send()` will guaranteedly send data over TLS, while `net_tls_connect()`, named similarly, won'y necessary make a TLS connect. Why this inconsistency and confusion?